packages/frames.js/src/getAddressForFid.ts

```diff
@@ -1,45 +1,47 @@
-import { VerificationAddEthAddressMessage } from "@farcaster/core";
 import { createPublicClient, http, parseAbi } from "viem";
 import { optimism } from "viem/chains";
-import { AddressReturnType } from "./types";
-import { bytesToHexString } from "./utils";
+import { AddressReturnType, HubHttpUrlOptions } from "./types";
 
 /**
  * Returns the first verified address for a given `Farcaster` users `fid` if available, falling back to their account custodyAddress
  */
 export async function getAddressForFid<
-  Options extends { fallbackToCustodyAddress?: boolean } | undefined,
+  Options extends
+    | ({ fallbackToCustodyAddress?: boolean } & HubHttpUrlOptions)
+    | undefined,
 >({
   fid,
-  hubClient,
-  options = { fallbackToCustodyAddress: true },
+  options = {},
 }: {
   /** the user's Farcaster fid, found in the FrameActionPayload message `fid` */
   fid: number;
-  /** A client for interacting with Farcaster hubs, of type HubRPCClient */
-  hubClient: any;
   options?: Options;
 }): Promise<AddressReturnType<Options>> {
-  const verificationsResult = await hubClient.getVerificationsByFid({
-    fid,
-  });
-  const verifications = verificationsResult.unwrapOr(null);
-  if (verifications?.messages[0]) {
+  // Merge default options with user provided options
+  options = {
+    fallbackToCustodyAddress: options.fallbackToCustodyAddress ?? true,
+    hubHttpUrl: options.hubHttpUrl ?? "https://nemes.farcaster.xyz:2281",
+  };
+
+  const verificationsResponse = await fetch(
+    `${options.hubHttpUrl}/v1/verificationsByFid?fid=${fid}`
+  );
+  const { messages } = await verificationsResponse.json();
+  if (messages[0]) {
```

Comment:
what if messages is the empty array? won't this crash?

Reply:
if messages is an empty array messages[0] will be undefined

```
Welcome to Node.js v20.11.0.
Type ".help" for more information.
> const messages = []
undefined
> messages[0]
undefined
> 
```